socket_test.go

```diff
@@ -322,3 +322,68 @@ func TestConnMaxRetriesInfinite(t *testing.T) {
 		t.Fatalf("Dial called %d times, expected  at least %d", transport.dialCalledCount, atLeastExpectedRetries)
 	}
 }
+
+func TestSocketAutomaticReconnect(t *testing.T) {
+	listenEndpoint := "tcp://*:1234"
+	dialEndpoint := "tcp://localhost:1234"
+	message := "test"
+
+	ctx, cancel := context.WithCancel(context.Background())
+
+	wg := new(sync.WaitGroup)
+	defer wg.Wait()
+	defer cancel()
```

Comment:
for consistency, I'd move the `defer cancel()` closer to the `WithCancel` stanza.